tldr/cross_file_calls/parsers/java.py

```
@@ -0,0 +1,174 @@
+import os
+import re
+from typing import Dict, List, Optional
+
+from tldr.cross_file_calls.parsers.base import BaseParser
+from tldr.cross_file_calls.core import HAS_JAVA_PARSER, _get_java_parser
+
+
+class JavaParser(BaseParser):
+    """Parser for Java files."""
+    
+    def extract_calls(self, file_path: str, timeout: Optional[float] = None) -> List[Dict]:
```

Ruff (0.14.11)

Unused method argument: `timeout`

(ARG002)